src/corehost/inc/pal.h

```diff
@@ -96,7 +98,7 @@ namespace pal
 
     bool realpath(string_t& path);
     bool file_exists(const string_t& path);
-    std::vector<pal::string_t> readdir(const string_t& path);
+    void readdir(const string_t& path, std::vector<pal::string_t>* list);
```

Comment:
@gkhanna79, references are const __in params. Here path is const which makes it an __in param. Pointers are for __inout and __out params. There are other places in the current code which don't follow these, I'll fix them as I touch those functions.